crates/driver/src/domain/mempools.rs

```diff
@@ -143,6 +147,7 @@ impl Mempools {
                     }
                 }
             }
+            panic!("Block stream finished unexpectedly");
```

Comment:
This should only be an `error` log IMO. We'll learn about that via alerts as well but it doesn't crash the pod unnecessarily.

Reply:
In my previous comment, I was thinking about canceling the settlement and returning a meaningful error. Panic doesn't make much sense here IMO. I think we will get alerted if all the settlements fail for some period of time.

Reply:
Thanks guys for the discussion! I removed the panic :ok_hand: